### ISSUE TITLE
Error in Path.js function _addSquareCap

### ISSUE THREAD

**Issue by sapics** [paperjs/paper.js#850]
I had an error at https://github.com/paperjs/paper.js/blob/develop/src/path/Path.js#L2871.
`loc = segment.getLocation()` is null, thus, `normal = loc.getNormal().multiply(radius)` causes error.

**Comment by lehni** [comment]
What's the context of this occurring? Could you maybe provide some path data and the commands required to reproduce this?

**Comment by sapics** [comment]
I am so sorry, I lost data. When I get error, I will post.
This error was occurred when I create path with pencil like tool.

**Comment by lehni** [comment]
Any chance to reproduce this? I'd love to get this fixed...

**Comment by sapics** [comment]
Sorry for late response, i missed.
I have tried for Error reproduction, but it have not achieved yet...

It might be come from following reason.
In my old program, I have treated path._segments directly in some special case.
But, in my current program, I have always use path.setSegments for segments modification.

Modifying path._segments directly would be discouraged...
So, this issue might be better to close.
If i success to reproduce the error, i wish to reopen this issue.
